### PR TITLE
Feature/num create body height tests

### PR DESCRIPTION
**Pull request by BirgitSaalfeld**
Dear Severin, 
please check my tests! 
I introduced tests for BodyHeight. 
Best regards
Birgit

### CODE REVIEW

**Comment by BirgitSaalfeld** [comment]
@wlad Dear Wlad, for some reason my code fails in the "build-and-test-fhirbridge"-Ci with "OS Errror". What does it mean? What do I have to do?
Best regards
Birgit

**Comment by wlad** [comment]
@BirgitSaalfeld Hi Birgit, sorry for the delay. I'm missing Github notifications in 99% of cases.  If possible contact me via Slack or like you did as the last resort by forwarding the issue to  Birger. 

I'm checking the issue and will give you feedback asap.

**Comment by wlad** [comment]
@BirgitSaalfeld some resource files used by Robot tests were renamed. I'll try to figur out which ones and will push a fix.

image

**Comment by wlad** [comment]
@BirgitSaalfeld you can access Robot logs of each build in the 'Artifacts' tab
image


image

**Comment by BirgitSaalfeld** [comment]
@wlad 
Thank you for the explanation. The checks still fail, but now a double ".json" file end seems to be the reason. 

How can I change the files used for the check?
jsonjson

**Comment by wlad** [comment]
@BirgitSaalfeld yep I saw it, that was a typo. Just pushed another fix. Please note that I'll have to push another commit reactivating all Robot test. This is just a test CI run at the moment.

**Comment by wlad** [comment]
@BirgitSaalfeld I've just reactivated all tests. Let's see if the 15 previously failing are fixed now :)

**Comment by BirgitSaalfeld** [comment]
@wlad 
Glad you saw it =D
Another two questions: 
1) What is your name on slack? I can't find you
2) Do I have to add further files for more mappings in `tests/robot/OBSERVATION/01_create.robot` manually? Or does it usually work automatically?

**Comment by wlad** [comment]
> 1. What is your name on slack? I can't find you

Wladislaw Wagner in Slack of vitagroup. 

> 2. Do I have to add further files for more mappings in `tests/robot/OBSERVATION/01_create.robot` manually? Or does it usually work automatically?

No, you don't have to. Just be aware when you rename any files in `src/test/resources` that they might be in use by some Robot tests.

**Comment by wlad** [comment]
ah damn seems I've missed one. I'm arleady on it :)
image

**Comment by wlad** [comment]
@BirgitSaalfeld I've just pushed another fix. Fingers crossed it works this time :)

File: src/test/java/org/ehrbase/fhirbridge/fhir/observation/BodyHeightIT.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    @Test
+    void createBodyHeightSnomedPeriod() throws IOException {
+        create("create-body-height-snomed-period.json");
+    }
+
+    // #####################################################################################
+    // check payload
+    @Test
+    void mapping_normal() throws IOException {
```
Comment (SevKohler) [inline src/test/java/org/ehrbase/fhirbridge/fhir/observation/BodyHeightIT.java]: i changed something here, now a testMapping method is to be overwritten. Take a look on the develop branch.

File: src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java [hunk 2]
```diff
@@ -31,29 +34,34 @@ public KorpergrosseComposition toComposition(Observation observation) {
...
+
+        ZonedDateTime fhirEffectiveDateTime = null;
         try {
-            // default for every observation
-            fhirEffectiveDateTime = observation.getEffectiveDateTimeType();
 
-            //BSa Wie erscheint das im Mapping? Und wo finde ich das Referenzmodell mit diesen Infos zu openEHR?
-            grosseLangeObservation.setTimeValue(fhirEffectiveDateTime.getValueAsCalendar().toZonedDateTime());
-            grosseLangeObservation.setOriginValue(fhirEffectiveDateTime.getValueAsCalendar().toZonedDateTime()); // mandatory
+            if(observation.hasEffectiveDateTimeType()) {
```
Comment (SevKohler) [inline src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java]: Move into seperate methdo pls

File: src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java [hunk 3]
```diff
@@ -31,29 +34,34 @@ public KorpergrosseComposition toComposition(Observation observation) {
...
             grosseLangeObservation.setLanguage(Language.DE); // FIXME: we need to grab the language from the template
             grosseLangeObservation.setSubject(new PartySelf());
 
             // special mapping content
             grosseLangeObservation.setGrosseLangeUnits(observation.getValueQuantity().getCode());
-            //BSa Hier ist nicht sichtbar, ob Körpergröße oder Geburt gesetzt wird -> woher die Info?
+
             grosseLangeObservation.setGrosseLangeMagnitude(observation.getValueQuantity().getValue().doubleValue());
 
         } catch (Exception e) {
```
Comment (SevKohler) [inline src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java]: Do you really want to catch every exception here or just specific ones ?

**Review by SevKohler — CHANGES_REQUESTED** [review]

**Comment by SevKohler** [comment]
@BirgitSaalfeld if you already change the things i mentioned above, you can fix your single code smell, then you produced zero new smells :)

File: src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java (hunk 3 above)
Comment (BirgitSaalfeld) [reply src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java]: That was overtaken from our "gold standard" example body weight. I only would forward the exception, so I think I can remove the try-catch block. What do you think?

File: src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java (hunk 2 above)
Comment (BirgitSaalfeld) [reply src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java]: I made many more methods and I wonder how you like it. The try-catch from the second comment you gave me is removed now. Is that ok?

File: src/test/java/org/ehrbase/fhirbridge/fhir/observation/BodyHeightIT.java (hunk 1 above)
Comment (BirgitSaalfeld) [reply src/test/java/org/ehrbase/fhirbridge/fhir/observation/BodyHeightIT.java]: Jep, much better code now :)

**Comment by BirgitSaalfeld** [comment]
@SevKohler There is one code smell left and I don't know how to fix it. I looked in the RadiologyIT and there it seems to be the same. Do you have any idea?

**Review by SevKohler — APPROVED** [review]

**Pull request merged**